general/array.hpp

```diff
@@ -565,17 +561,51 @@ inline void Swap(Array<T> &a, Array<T> &b)
 {
    Swap(a.data, b.data);
    Swap(a.size, b.size);
-   Swap(a.allocsize, b.allocsize);
-   Swap(a.inc, b.inc);
+}
+
+template <class T>
+inline Array<T>::Array(const Array &src)
+   : size(src.Size())
+{
+   size > 0 ? data.New(size, src.data.GetMemoryType()) : data.Reset();
+   data.CopyFrom(src.data, size);
+   data.SetExecFlag(src.data.GetExecFlag());
+}
+
+template <typename T> template <typename CT>
+inline Array<T>::Array(const Array<CT> &src)
+   : size(src.Size())
+{
+   size > 0 ? data.New(size) : data.Reset();
+   for (int i = 0; i < size; i++) { (*this)[i] = T(src[i]); }
+}
+
+template <class T>
+inline void Array<T>::GrowSize(int minsize)
+{
+   const int nsize = std::max(minsize, 2 * data.Capacity());
+   Memory<T> p(nsize, data.GetMemoryType());
+   p.CopyFrom(data, size);
+   p.SetExecFlag(data.GetExecFlag());
+   data.Delete();
+   data = p;
+}
+
+template <typename T> template <typename CT>
+inline Array<T> &Array<T>::operator=(const Array<CT> &src)
+{
+   SetSize(src.Size());
+   for (int i = 0; i < size; i++) { (*this)[i] = T(src[i]); }
```

Comment:
Should we get the pointers out first to avoid extra function calls?

Reply:
These are inline methods.